cord19-explorer/src/main/java/edu/asu/diging/cord19/explorer/web/auth/CoordinateCleaningController.java

```diff
@@ -0,0 +1,134 @@
+    
+    @Autowired
+    private PublicationRepository pubRepo;
+    
+    @Autowired
+    private MongoTemplate mongoTemplate;
+
+    
+    @RequestMapping(value = "/auth/coordinates/clean", method = RequestMethod.GET)
+    public String show() {
+
+        return "auth/cleanCoordinates";
+    }
+    
+    @RequestMapping(value = "/auth/coordinates/clean", method = RequestMethod.POST)
+    public String start() throws ClassCastException, ClassNotFoundException, IOException {
+        try (CloseableIterator<PublicationImpl> docs = mongoTemplate.stream(new Query(), PublicationImpl.class)) {
+            while (docs.hasNext()) {
+                PublicationImpl pub = docs.next();
+                List<PersonImpl> authors = pub.getMetadata().getAuthors();
+                for(PersonImpl author : authors) {
+                    List<String> coords = new ArrayList();
+                    List<Double> formattedCoords = new ArrayList();
+                    if(author.getAffiliation().getSelectedWikiarticle() != null) {
+                       WikipediaArticleImpl article = author.getAffiliation().getSelectedWikiarticle();
+                       String [] splitted =  article.getCoordinates().split("\\|");
+                       // split on pipes then use Ordinals to gather Degree, Mins, Secs then compute Decimal Degrees
+                       //Refactor into function
+                       for (String s: splitted) {      
+                            if(s.matches("-?\\d+(\\.\\d+)?")) {
+                                System.out.println(s);
+                                coords.add(s);
+                                
+                            } else if(s.equals("N") || s.equals("E")){
+                                if(coords.size() == 3) {
+                                    float d = Float.parseFloat(coords.get(0));
+                                    float m = Float.parseFloat(coords.get(1));
+                                    float se = Float.parseFloat(coords.get(2));
+                                    double dd = 0;
+                                    if(d == 0) {
+                                        dd = ((m / 60.0) + (se / 3600.0));
+                                    } else {
+                                        dd = Math.signum(d) * (Math.abs(d) + (m / 60.0) + (se / 3600.0));
+                                    }
+                                    formattedCoords.add(dd);
+                                    coords.clear();
+                                }
+                                if(coords.size() == 2) {
+                                    float d = Float.parseFloat(coords.get(0));
+                                    float m = Float.parseFloat(coords.get(1));
+                                    double dd = Math.signum(d) * (Math.abs(d) + (m / 60.0));
+                                    formattedCoords.add(dd);
+                                    coords.clear();
+                                }
```

Comment:
if size = 2 and size =3 can be combined, can't they?
```
if(coords.size() >= 2) {
    float d = Float.parseFloat(coords.get(0));
    float m = Float.parseFloat(coords.get(1));
    float se = coords.size() => 3 ? se = Float.parseFloat(coords.get(2)) : 0;
    double dd = Math.signum(d) * (Math.abs(d) + (m / 60.0) + (se / 3600.0));
    formattedCoords.add(dd);
 }
```
and define coords within the loop so you don't have to clear it.